python/dgl/graph.py

```diff
@@ -385,6 +385,15 @@ def clear(self):
         self._msg_graph.clear()
         self._msg_frame.clear()
 
+    def clear_cache(self):
```

Comment:
I think it's better to make the name very specific for clearing cache for graph structures. Maybe passing an argument?

Reply:
How about we adding extra option arguments when we have more types of caches in the future? Currently, I'm not sure what are the other cache types, and I want to keep it the API as simple as possible at the moment.

Reply:
i think the main concern is back compatibility. Adding an argument in the future is kind of breaking API.

I'm sure we'll have other cache types. What I'm working on is the GPU caching for node embeddings. We'll need to provide caching in the main memory in the distributed setting.

Reply:
Breaking API at the current stage is fine. This is definitely not the final API in our next release. For distributed setting, we will likely to have extra APIs (similar to DataParallelTrainer). We could put your caching option there. I would suggest modify this API again (if necessary) in the PR for distributed training, so we can have the whole picture to look at.

Reply:
sounds good.